cmd/group-auto-updater/main_test.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	coretesting "k8s.io/client-go/testing"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/diff"
+
+	v1 "github.com/openshift/api/user/v1"
+	"github.com/openshift/client-go/user/clientset/versioned/fake"
+)
+
+func TestUpdateGroup(t *testing.T) {
+	peribolosConfig := []byte(`orgs:
+  test-org:
+    admins:
+    - admin1
+    - admin2
+    - admin3
+    - admin4
+    members:
+    - member1
+    - member2`)
+
+	tempDir, err := ioutil.TempDir("", "test")
+	if err != nil {
+		t.Fatalf("Unexpected error creating tmpdir: %v", err)
+	}
+	defer os.RemoveAll(tempDir)
+
+	peribolosConfigPath := filepath.Join(tempDir, "peribolos-config.yaml")
+	if err = ioutil.WriteFile(peribolosConfigPath, peribolosConfig, 0664); err != nil {
+		t.Fatalf("Unexpected error in writing peribolos config: %v", err)
+	}
+
+	testGroup := "test-group"
+	testOrg := "test-org"
+
+	baseGroup := &v1.Group{
+		ObjectMeta: metav1.ObjectMeta{Name: "test-group"},
+		Users:      v1.OptionalNames{"user1", "user2"},
+	}
+
+	expectedGroup := &v1.Group{
+		ObjectMeta: metav1.ObjectMeta{Name: "test-group"},
+		Users:      v1.OptionalNames{"admin1", "admin2", "admin3", "admin4", "member1", "member2"},
+	}
+	fakecs := fake.NewSimpleClientset()
+	fakecs.PrependReactor("update", "groups", func(action coretesting.Action) (bool, runtime.Object, error) {
+		group := action.(coretesting.UpdateAction).GetObject().(*v1.Group)
+		if group.Name == testGroup {
+			if !equality.Semantic.DeepEqual(group, expectedGroup) {
+				t.Fatalf(diff.ObjectReflectDiff(group, expectedGroup))
+			}
+		} else {
+			t.Fatalf("unknown group found: %#v", group)
+		}
+
+		return false, nil, nil
+	})
+
+	userV1Client := fakecs.UserV1()
+	if _, err := userV1Client.Groups().Create(baseGroup); err != nil {
+		t.Fatalf("%v", err)
+	}
+
+	users, err := getUsersFromPeribolosConfig(peribolosConfigPath, testOrg)
+	if err != nil {
+		t.Fatalf("coulnd't get users from peribolos config: %v", err)
+	}
+
+	if err := updateGroup(testGroup, users, userV1Client.Groups()); err != nil {
```

Comment:
We should also check that the API calls were actually made (at least the update one).